typescript/src/schema/naming.ts

```diff
@@ -32,9 +42,13 @@ export class Naming {
     // common prefix must either end with `.`, or be equal to at least one of
     // the packages' prefix
     if (!prefix.endsWith('.') && !protoPackages.some(pkg => pkg === prefix)) {
-      throw new Error('Protos provided have different proto packages.');
-    }
-    const rootPackage = prefix.replace(/\.$/, '');
+      if (mainServiceName) {
+        rootPackage = this.checkServiceInPackage(
+          protoPackages,
+          mainServiceName
+        );
+      } else throw new Error('Protos provided have different proto packages.');
```

Comment:
Please always use braces `{ }` after if/else. Consequences of not using them can be disastrous :) https://blog.codecentric.de/en/2014/02/curly-braces/